ddr/ddr/Models/Leveling/Spawner.swift

```diff
@@ -0,0 +1,68 @@
+
+import Foundation
+import SpriteKit
+
+protocol Spawnable: Hashable {
+  func despawn()
+}
+
+class Spawner<T: Spawnable>{
+  
+  private let maxSpawned: Int
+  private let minSpawned: Int
+  private let maxConcurrent: Int
+  private let cooldown: TimeInterval
+  private let pSpawn: CGFloat
+  private let getNewSpawn: (Spawner) -> T
+  
+  private var canSapwnAfter: Date
```

Comment:
typo - canSpawnAfter

Reply:
Wow, the power of tab-complete. Fixed in next commit.